avro/src/main/scala/hydra/avro/registry/SchemaRegistry.scala

```diff
@@ -203,14 +199,34 @@ object SchemaRegistry {
           schemas <- versions.traverse(getSchemaFor(subject, _)).map(_.flatten).checkKeyEvolution(subject, schema)
           validated <- Sync[F].pure(validate(schema, schemas.reverse))
           _ <- checkLogicalTypesCompat(schema)
+          latest <- getLatestSchemaBySubject(subject)
           schemaVersion <- if (validated) {
+            if(checkForOnlyDocFieldUpdate(schema, latest)) {
```

Comment:
Out of curiousity, why don't you do a if (validated && checkForOnlyDocFieldUpdate(schema, latest)) instead of nested ifs?

Reply:
I always want it to run schemaRegistryClient.register(), but I only want it to reset if the nested condition is met.